<launch>
  <arg name="machine" default="localhost" />
  <arg name="user" default="" />
  <arg name="timeout" default="1.0" />
  <arg name="max_dist" default="4.0" />
  <arg name="min_dist" default="1.0" />
  <arg name="detection_angle" default="45.0" />
  <arg name="gaze_type" default="0" />
  <arg name="with_site_movebase_params" default="false"/>
  <arg name="site_movebase_params" default=""/>
  <arg name="people_positions" default="/people_tracker_filter/positions"/>
  <arg name="DWAPlannerROS_srv" default="/move_base/DWAPlannerROS"/>
  <arg name="DWAPlannerROS_han_srv" default="/human_aware_navigation/DWAPlannerROS"/>
  <arg name="simple_goal_topic" default="/move_base_simple/goal"/>


  <arg name="moveBase_ac" default="move_base"/>
  <arg name="useGaze" default="true"/>

  <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER)" user="$(arg user)" default="true"/>
  
  <node pkg="strands_human_aware_navigation" type="human_aware_navigation.py" name="human_aware_navigation" output="screen" respawn="true">
    <param name="timeout" value="$(arg timeout)" type="double"/>
    <param name="max_dist" value="$(arg max_dist)" type="double"/>
    <param name="min_dist" value="$(arg min_dist)" type="double"/>
    <param name="detection_angle" value="$(arg detection_angle)" type="double"/>
    <param name="gaze_type" value="$(arg gaze_type)" type="int"/>
    <param name="people_positions" value="$(arg people_positions)" type="string"/>
    <param name="DWAPlannerROS_srv" value="$(arg DWAPlannerROS_srv)" type="string"/>
    <param name="moveBase_ac" value="$(arg moveBase_ac)" type="string"/>
    <param name="simple_goal_topic" value="$(arg simple_goal_topic)" type="string"/>

    <param name="useGaze" value="$(arg useGaze)" type="bool"/>
  </node>


  <node pkg="strands_human_aware_navigation" type="dwa_dyn_wrapper" name="DWAPlannerROS_dyn_wrapper" output="screen" respawn="true">
    <param name="human_aware_navigation/DWAPlannerROS/DWAPlannerROS_srv" value="$(arg DWAPlannerROS_srv)" type="string"/>
    <param name="human_aware_navigation/DWAPlannerROS/DWAPlannerROS_han_srv" value="$(arg DWAPlannerROS_han_srv)" type="string"/>
  </node>

  <rosparam file="$(find strands_movebase)/strands_movebase_params/dwa_planner_ros.yaml" command="load" ns="human_aware_navigation"/>
  <!-- THIS SHOULD STAY LAST, so that sites can override anything. -->
  <rosparam file="$(arg site_movebase_params)" command="load" if="$(arg with_site_movebase_params)" ns="human_aware_navigation"/>
</launch>
